### PR TITLE
Add hash implementation of signing traits

### PR DESCRIPTION
**Pull request by ltseeley**
Adds the `HashContext`, `HashSigner`, and `HashVerifier` behind the
`hash` trait. This implementation is intended for testing purposes.

Signed-off-by: Logan Seeley <seeley@bitwise.io>

### CODE REVIEW

File: libcylinder/Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,7 @@ experimental = [
   # The following features are experimental:
 ]
 
+hash = ["sha2"]
```
Comment (peterschwarz) [inline libcylinder/Cargo.toml:41]: This feature should be under stable or  experimental, presumably

Comment (ltseeley) [reply libcylinder/Cargo.toml:42]: Added to experimental

File: libcylinder/tests/mod.rs [hunk 2]
```diff
@@ -21,6 +21,22 @@ use cylinder::{Context, PrivateKey, Signature};
 
 type TestResult = Result<(), Box<dyn std::error::Error>>;
 
+/// Verifies that the round-trip of signing and verifying some bytes using a signer and verifier
+/// produced using the given context and private key.
+fn test_signing_and_verification<C: Context + 'static>(
```
Comment (peterschwarz) [inline libcylinder/tests/mod.rs:26]: This is a good test, can you add the test run to the secp256k1 tests, too?

Comment (ltseeley) [reply libcylinder/tests/mod.rs:26]: I have it ready in a follow-up PR

**Pull request merged**